src/solvers/prop/prop_conv.h

```diff
@@ -128,7 +128,7 @@ class prop_conv_solvert:public prop_convt
   bool post_processing_done;
 
   // get a _boolean_ value from counterexample if not valid
-  virtual bool get_bool(const exprt &expr, tvt &value) const;
+  virtual optionalt<tvt> get_bool(const exprt &expr) const;
```

Comment:
Again; this doesn't seem broken and so doesn't need to be fixed.  If someone gives a true expression in then it is perfectly OK to give a true literal back.

Reply:
Again, this doesn't change the behavior. The interface as it was before is confusing because you would think that the result of `get_bool` is the returned `bool`, while in fact it is stored in `tvt &value` (your comment makes me think you are confused yourself). Using optional makes it much clearer that it's actually a `tvt` which is the result.

(note to myself) the documentation should be updated.